### PR TITLE
Catalog Modal Only

### PR DESCRIPTION
**Pull request by gilhsu**
- Create New `CatalogModal`
- Access `CatalogModal` from `CatalogScreen` upper right icon
- Create `showCollectionOnly` state slice
- Create `showMarkedOnly` state slice
- Introduce `SegmentControl` to toggle these new state slices

https://user-images.githubusercontent.com/45667486/113917159-dcd54480-97ae-11eb-86b9-58b5058ee4f5.mov

### CODE REVIEW

File: src/redux/reducers/index.js [hunk 1]
```diff
@@ -62,16 +62,11 @@ export const resourceTypeFiltersReducer = (state = preloadResourceTypeFilters, a
   switch (action.type) {
     case actionTypes.CREATE_COLLECTION:
     case actionTypes.SELECT_COLLECTION:
+    case actionTypes.DELETE_COLLECTION:
+    case actionTypes.CLEAR_COLLECTION:
```
Comment (gilhsu) [inline src/redux/reducers/index.js:66]: this is cleanup from the pivot to consolidating `CollectionPreview` with `CatalogScreen`

File: src/redux/reducers/index.js [hunk 2]
```diff
@@ -104,16 +99,11 @@ export const dateRangeFilterReducer = (state = preloadSelectedTimelineRange, act
...
```
Comment (gilhsu) [inline src/redux/reducers/index.js:103]: this is cleanup from the pivot to consolidating `CollectionPreview` with `CatalogScreen`

File: src/components/SegmentControl/CollectionSegmentControl.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+const allRecordsDescription = 'Displays all records.';
+const collectionRecordsDescription = 'Only displays records saved to the collection.';
+
+const CollectionSegmentControl = ({
+  showCollectionOnly,
+  toggleShowCollectionOnlyAction,
+}) => {
+  const segControlIndex = showCollectionOnly ? 1 : 0;
+  const description = segControlIndex === 0 ? allRecordsDescription : collectionRecordsDescription;
+  const handleChange = (event) => {
```
Comment (foxtree) [inline src/components/SegmentControl/CollectionSegmentControl.js]: should this be `selectedSegmentIndex`?

File: src/components/SegmentControl/CollectionSegmentControl.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    if (event === 0) {
+      toggleShowCollectionOnlyAction(false);
+    } else {
+      toggleShowCollectionOnlyAction(true);
+    }
```
Comment (foxtree) [inline src/components/SegmentControl/CollectionSegmentControl.js]: maybe: 
`toggleShowCollectionOnly(selectedSegmentIndex !== 0);`

File: src/components/SegmentControl/MarkedSegmentControl.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+const allRecordsDescription = 'Displays all records.';
+const highlightedRecordsDescription = 'Only displays highlighted records.';
+
+const MarkedSegmentControl = ({
+  showMarkedOnly,
+  toggleShowMarkedOnlyAction,
+}) => {
+  const segControlIndex = showMarkedOnly ? 1 : 0;
+  const description = segControlIndex === 0 ? allRecordsDescription : highlightedRecordsDescription;
+  const handleChange = (event) => {
```
Comment (foxtree) [inline src/components/SegmentControl/MarkedSegmentControl.js]: selectedSegmentIndex

File: src/components/SegmentControl/MarkedSegmentControl.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+      toggleShowMarkedOnlyAction(true);
+    }
+  };
+
+  return (
+    <View style={styles.root}>
+      <BaseSegmentControl
+        values={['All Records', 'Highlighted Records']}
+        selectedIndex={segControlIndex}
+        handleChange={handleChange}
```
Comment (foxtree) [inline src/components/SegmentControl/MarkedSegmentControl.js]: not 100% of latest conventions, but this possibly should be `onChange={handleChange}`

eg, there's the following ESLint rule: (which may not be configured for this repo)

https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-handler-names.md

**Review by foxtree — APPROVED** [review]

**Pull request merged**